elasticsearch_dsl/utils.py

```diff
@@ -110,18 +122,20 @@ def to_list(self):
         return self._l_
 
 
-class AttrDict:
+class AttrDict(Generic[_KeyT, _ValT]):
     """
     Helper class to provide attribute like access (read and write) to
     dictionaries. Used to provide a convenient way to access both results and
     nested dsl dicts.
     """
 
-    def __init__(self, d):
+    _d_: Dict[_KeyT, _ValT]
+
+    def __init__(self, d: Dict[_KeyT, _ValT]):
         # assign the inner dict manually to prevent __setattr__ from firing
         super().__setattr__("_d_", d)
 
-    def __contains__(self, key):
```

Comment:
Would this type be `_KeyT` instead of `object`?

Reply:
I kept is as `object` to be consistent with the `dict.__contains__()` typing, conceptually I think is also not a type error to ask if an object of a wrong type is in the dictionary, is just gonna be false.